synapse_spamcheck_badlist/bad_list_filter.py

```diff
@@ -85,16 +87,20 @@ def __init__(self, config, api):
         self._link_automaton = None
 
         # Start the loop to update links.
-        self._update_links_loop = LoopingCall(
-            lambda: defer.ensureDeferred(self._update_links_automaton())
+        api.looping_background_call(
+            f=self._update_links_automaton,
+            msec=pull_from_db_every_sec * 1000,
+            desc="Background update list of bad links",
         )
-        self._update_links_loop.start(pull_from_db_every_sec, now=False)
+
         # As soon as we can, run the first fetch.
         # Note that we have no guarantee that this is finished
         # by the time we receive the first message, so we need
         # a fallback in `_get_links_automaton()`.
         reactor.callWhenRunning(
-            lambda: defer.ensureDeferred(self._update_links_automaton())
+            lambda: defer.ensureDeferred(
+                context.run_in_background(self._update_links_automaton)
```

Comment:
this ought to be `run_as_background_process`: `run_in_background` is only appropriate when there is an existing logcontext that will be active for the duration of the background task (ie, it's useful for parallelisation, but not so much for discrete tasks)